src_python/habitat_sim/simulator.py

```diff
@@ -612,23 +612,22 @@ def step(
 
         for agent_id, agent_act in action.items():
             agent = self.get_agent(agent_id)
-            self.agents[agent_id].did_collide = agent.act(agent_act)
+            agent.act(agent_act)
             self.__last_state[agent_id] = agent.get_state()
 
         # step physics by dt
         step_start_Time = time.time()
         super().step_world(dt)
         self._previous_step_time = time.time() - step_start_Time
 
-        agent_ids = list(action.keys())
-        per_agent_observations = self.get_sensor_observations(agent_ids)
-
+        per_agent_observations = self.get_sensor_observations(list(action.keys()))
```

Comment:
```suggestion
per_agent_observations = self.get_sensor_observations(action.keys())
```
Does the above not work?

Reply:
Ah nvm, found your earlier comment.